modules/dmrpp_module/DMZ.cc

```diff
@@ -132,13 +150,27 @@ static inline DmrppCommon *dc(BaseType *btp)
     return dc;
 }
 
+
+/**
+ * Loads configuration state from TheBESKeys
+ *
+ */
+void DMZ::load_config_from_keys()
+{
+    // ########################################################################
+    // Loads the ELIDE_UNSUPPORTED_KEY (see top of file for key definition)
+    // And if it's set, and set to true, then we set the eliding flag to true.
+    d_elide_unsupported = TheBESKeys::TheKeys()->read_bool_key(ELIDE_UNSUPPORTED_KEY,false);
+}
+
 /**
  * @brief Build a DMZ object and initialize it using a DMR++ XML document
  * @param file_name The DMR++ XML document to parse.
  * @exception BESInternalError if file_name cannot be parsed
  */
 DMZ::DMZ(const string &file_name)
 {
+    load_config_from_keys();
```

Comment:
I'm in a quandary about constructors that do things that can throw, esp. when those exceptions are not fatal. Actually, more and more, I think we should rarely have ctors other than the defaults or sones that just pass strings, ints, etc., onto members.

We could drop this ctor and say that to use this class, you have to load the keys or get undefined behavior (or we could put calling load...() inside an init() method. Because...

In my PR, I add another way to parse, so if the DMZ is made using the default ctor and then parse_xml_string(...) is called, the keys will never be loaded. But, while there are ways around this particular problem, it's a general issue. Code that throws should not be in a ctor. I did this all over the place in the past.